rueppellii-lvlup-asp.net-core/Controllers/BadgesController.cs

```diff
@@ -11,27 +11,21 @@ namespace rueppellii_lvlup_asp.net_core.Controllers
     [Authorize]
     public class BadgesController : Controller
     {
-    private readonly ObjectContainer objectContainer;
-    private readonly IMapper mapper;
+        private readonly ICrudService<BadgeDto, Badge> service;
 
-    public BadgesController(IMapper mapper)
-    {
-
-      objectContainer = new ObjectContainer();
-      objectContainer.Badges[0] = new LevelDto() { Name = "Process Improver", Level = 2 };
-      objectContainer.Badges[1] = new LevelDto() { Name = "English Speaker", Level = 1 };
-      objectContainer.Badges[2] = new LevelDto() { Name = "Feedback Giver", Level = 1 };
-      this.mapper = mapper;
-    }
-
-    [HttpGet("badges")]
-    public IActionResult ListBadges()
-    {
-        if (Request.Headers["usertokenauth"] == "gen")
+        public BadgesController(ICrudService<BadgeDto, Badge> service)
         {
-        return Ok(objectContainer);
+            this.service = service;
         }
-        return Unauthorized(new ErrorMessage("Unauthorized"));
+
+        [HttpGet("badges")]
+        public IActionResult ListBadges()
+        {
+            if (Request.Headers["usertokenauth"] == "gen")
+            {
+                return Ok(service.GetAll());
+            }
```

Comment:
The else is not mandatory for this logic to be executed

Reply:
👍